### DISCUSSION TITLE
EAGE-HACK-2024: Helpful Packages, Code, Data, and Technology

### DISCUSSION THREAD

**Discussion by aruss175** [EAGE-Annual-Hackathon/EAGE-Annual-Hackathon-Discussion-Boards#13] [Ideas]
Here you will find tips, tricks, links to data, links to tools and guides to help access these things.

**Comment by aruss175** [comment]
Here is a starting point of data and technology that could be used in this hackathon:
----------------------------------------------
**ResSim3D Simulator Reservoir simulator**
powered by NVIDIA
**Modulus** (developed by
Clement Etienam.
Physics-ML powered

https://github.com/NVIDIA/modulus-sym/tree/main/examples/reservoir_simulation

**Modulus examples**
https://github.com/NVIDIA/modulus-sym/tree/main

----------------------------------------------
**Norne Dataset Reservoir model** 
https://github.com/NVIDIA/modulus-sym/tree/main/examples/reservoir_simulation/Norne

----------------------------------------------

**Marmousi Dataset Synthetic subsurfacemodel for seismic wave propagation and inversion**

https://mondaic.com/docs/0.12.13/examples/tutorials/advanced_interface/marmousi/tutorial

----------------------------------------------

**Volve Dataset Field seismic data and model**

https://github.com/equinor/segyio-notebooks/blob/master/notebooks/pylops/01_seismic_inversion.ipynb

https://github.com/maribickpostanes/Interactive-Seismic-Data-Section-Plot
https://github.com/maribickpostanes/Interactive-Horizon-Surfaces-Plot
https://www.christianhaller.me/blog/projectblog/2021-02-01-Volve_ML/
https://github.com/maribickpostanes/Interactive-Well-Trajectory-Plot

----------------------------------------------

**Smeaheia Dataset** Subsurface data including wells, seismic geomechanics and simulation models

https://co2datashare.org/dataset/smeaheia-dataset

----------------------------------------------

**Records from the Norwegian Authorities' Database for reports on the Norwegian Continental Shelf**
https://zenodo.org/records/10775273

**Comment by aruss175** [comment]
**And Join us for a pre-hackathon workshop on Modulus Jun 7, 2024:**
Physics-ML with Modulus framework - Dr. Clement Etienam

1h 

NVIDIA Modulus, the physics machine learning platform, turbocharges such use cases by building physics-based deep learning models that are 100,000x faster than traditional methods and offer high-fidelity simulation results.

**Reply by roderickperez** [reply]
Is this workshop virtual?

**Reply by ovcharenkoo** [reply]
Hi @roderickperez, yes, all workshops are virtual. You will receive a link in an email closer to the event

**Comment by ovcharenkoo** [comment]
**As a starting point for LLM fine-tuning on geoscience data we provide:**
- A dataset and processing scripts [~800 EAGE abstracts](https://drive.google.com/drive/folders/1zw9-kiMypuyj09aWBXQKKtik3wIQsCRK?usp=sharing)
- Example workflow for [LoRA fine-tuning of Llama-2-7B model on this dataset using NeMo framework](https://docs.nvidia.com/nemo-framework/user-guide/latest/playbooks/llama2peft.html)

**Join us for a pre-hackathon workshop on NeMo Framework for LLM Training and Fine-tuning**
A dive into the essentials of Language Model training and fine-tuning. This workshop aims to walk you through examples and methods needed to enable the adaptation of an LLM to a new knowledge domain. By [Oleg Ovcharenko](https://www.linkedin.com/in/ovcharenkoo/?originalSubdomain=ae)

**Registered participants will receive and email a few weeks before the event with details about both workshops**

GitHub repository with hands-on tutorial on Llama-2-7B LoRA fine-tuning could be found [here](https://github.com/ovcharenkoo/nemo_lora_handson)
